deepchecks/nlp/text_data.py

```diff
@@ -347,6 +348,17 @@ def calculate_default_properties(self, include_properties: t.List[str] = None,
                                                   ignore_properties=ignore_properties)
         self._properties = pd.DataFrame(properties, index=self.index)
 
+    def set_properties(self, properties: pd.DataFrame):
+        """Set the properties of the dataset."""
+        if self._properties is not None:
+            warnings.warn('Properties already exist, overwriting them', UserWarning)
+
+        if not isinstance(properties, pd.DataFrame):
+            raise DeepchecksValueError('properties must be a pandas DataFrame')
+        if list(properties.index) != self.index:
+            raise DeepchecksValueError('properties index must be the same as the dataset index')
+        self._properties = properties
```

Comment:
1. Add set_metadata as well. 
2. **remove** the parallel code in the __init__ and just use this function in the __init__.
3. In context.assert_properties/metadata, tell the user they can use "set_x" functions to add these (instead / in addition to initialize the object)